ops/model.py

```diff
@@ -2509,7 +2509,15 @@ def remove_path(self, path: Union[str, PurePath], *, recursive: bool = False):
 
         Args:
             path: Path of the file or directory to delete from the remote system.
-            recursive: If True, recursively delete path and everything under it.
+            recursive: If True, and path is a directory, recursively deletes it and
+                       everything under it. If the path is a file, delete the file and
+                       do nothing if the file is non-existent. Behaviourally similar
+                       to ``rm -rf <file|dir>``.
```

Comment:
I realise this wasn't your wording, but maybe we can clean it up slightly while we're here. The "do nothing if the file is non-existent" sort of implies it'll only do nothing if it doesn't exist and it's a *file*, but the same is true for directories with `recursive=True`. Perhaps:

```suggestion
            recursive: If True, and path is a directory, recursively delete it and
                       everything under it. If path is a file, delete the file. In
                       either case, do nothing if the file or directory does not
                       exist. Behaviourally similar to ``rm -rf <file|dir>``.```